src/ray/raylet/node_manager.cc

```diff
@@ -1339,6 +1311,45 @@ void NodeManager::ProcessAnnounceWorkerPortMessage(
   if (is_worker) {
     worker_pool_.OnWorkerStarted(worker);
     HandleWorkerAvailable(worker);
+  } else {
+    // Driver is ready. Add the job to GCS.
+    JobID job_id = worker->GetAssignedJobId();
+    boost::optional<const rpc::JobConfig &> job_config =
+        worker_pool_.GetJobConfig(job_id);
+    RAY_CHECK(job_config.has_value());
+
+    rpc::Address driver_address;
+    // Assume raylet ID is the same as the node ID.
+    driver_address.set_raylet_id(self_node_id_.Binary());
+    driver_address.set_ip_address(worker->IpAddress());
+    driver_address.set_port(port);
+    driver_address.set_worker_id(worker->WorkerId().Binary());
+    auto job_data_ptr =
+        gcs::CreateJobTableData(job_id,
+                                /*is_dead=*/false,
+                                driver_address,
+                                worker->GetProcess().GetId(),
+                                string_from_flatbuf(*message->entrypoint()),
+                                *job_config);
+
+    RAY_CHECK_OK(gcs_client_->Jobs().AsyncAdd(job_data_ptr, [client](Status status) {
+      if (!status.ok()) {
+        RAY_LOG(ERROR) << "Failed to add job to GCS: " << status.ToString();
+      }
```

Comment:
In the original code, if this fails, `RegisterClientReply` will have 

```
// Whether the registration succeeded.
  success: bool;
  // The reason of registration failure.
  failure_reason: string;
```

set. We should do the same thing for AnnounceWorkerPortReply?

Reply:
ok, added reply status. But at the terminal we check-fail it, since the cython binding `notify_raylet` returns void. Do we want to also change that to a raise exception?